src/test/java/org/kiwiproject/dropwizard/util/health/keystore/ExpiringKeystoreHealthCheckTest.java

```diff
@@ -0,0 +1,267 @@
+
+            softly.assertThat(result.isHealthy()).isTrue();
+            softly.assertThat(result.getMessage()).isEqualTo(path);
+            softly.assertThat(result.getError()).isNull();
+
+            var details = result.getDetails();
+            softly.assertThat(details).contains(
+                    entry("path", path),
+                    entry("ttl", Duration.days(20))
+            );
+
+            List<BasicCertInfo> validCerts = getBasicCertInfoList(details, "validCerts");
+            softly.assertThat(validCerts).hasSize(1);
+
+            var certInfo = first(validCerts);
+            softly.assertThat(certInfo.getSubjectDN()).isEqualTo("CN=Valid Test, OU=Development, O=Project, L=There, ST=VA, C=US");
+            softly.assertThat(certInfo.getIssuerDN()).isEqualTo("CN=Valid Test, OU=Development, O=Project, L=There, ST=VA, C=US");
+
+            var sixtyDaysAgo = ZonedDateTime.now(ZoneId.of("UTC")).minusDays(60);
+            var sixtyDaysAgoString = DATE_FORMATTER.format(sixtyDaysAgo);
+            softly.assertThat(certInfo.getIssuedOn()).startsWith(sixtyDaysAgoString);
+
+            var expirationDate = sixtyDaysAgo.plusDays(600);
+            var expirationDateString = DATE_FORMATTER.format(expirationDate);
+            softly.assertThat(certInfo.getExpiresOn()).startsWith(expirationDateString);
+
+            List<BasicCertInfo> expiredCerts = getBasicCertInfoList(details, "expiredCerts");
+            softly.assertThat(expiredCerts).isEmpty();
+
+            List<BasicCertInfo> expiringCerts = getBasicCertInfoList(details, "expiringCerts");
+            softly.assertThat(expiringCerts).isEmpty();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static List<BasicCertInfo> getBasicCertInfoList(Map<String, Object> details, String key) {
+        return (List<BasicCertInfo>) details.get(key);
+    }
+
+    @Nested
+    class ToCertificate {
+        @Test
+        void shouldThrowException_WhenKeyStoreNotLoaded() throws KeyStoreException {
+            var keyStore = KeyStore.getInstance(KeyStoreType.PKCS12.name());
+            var path = "/path/to/cert.pkc512";
+
+            assertThatThrownBy(() -> ExpiringKeystoreHealthCheck.toCertificate("anAlias", path, keyStore))
+                    .isExactlyInstanceOf(ExpiringKeystoreHealthCheck.KeyStoreNotLoadedException.class)
+                    .hasMessage("Keystore at path %s not loaded", path);
+        }
+    }
+
+    private String createTemporaryKeystore(Path dir, int daysValid, String dn) throws InterruptedException {
+        var keystorePath = dir.resolve("temporary-keystore.jks").toAbsolutePath().toString();
+
+        var processHelper = new ProcessHelper();
+        var keystoreGenProcess = processHelper.launch("keytool", "-alias", "Test", "-genkey", "-keystore", keystorePath, "-keyalg", "RSA", "-validity", String.valueOf(daysValid), "-startdate", "-60d", "-storepass", "unittest", "-keypass", "unittest", "-dname", dn);
```

Comment:
Suggest splitting the arguments onto separate lines, such that pairs e.g. `-alias Test` are on the same line whereas "singletons" e.e. `-genkey` are on a separate line by themselves. Make sense? That way you can easily scan down to see the arguments and also easily see which are key/value pairs.